Add theme support

samporapeli opened the pull request:
Fixes issue #17. Includes Classic theme (as before) and new Dark theme.

samporapeli committed:
Add support for reload when caps lock is on

samporapeli committed:
Remove unnecessary semicolons

samporapeli committed:
Remove more unnecessary semicolons

samporapeli committed:
Add missing semicolons

samporapeli committed:
Merge remote-tracking branch 'upstream/master'

Fetch changes from original repository.

samporapeli committed:
Add support for themes

Also add dark theme. Todo: change theme from UI

samporapeli committed:
Add theming js file and dark theme css

samporapeli committed:
Add theme support

This commit adds support for themes and changing between them.
Currently Classic and Dark themes are available.

samporapeli committed:
Refactor code to use colors directly from theme object

As there was no need to save the colors to local variables,
using them directly from theme object.

samporapeli committed:
Change development port back to normal

samporapeli committed:
Clarify configuration of dark theme

Make theme customization use assigning to object variables instead
of construction parameters. This makes it easier to follow what
values are changed. There's still support for using parameters.

ilarikuoppala commented:
`dark.css` adds unneccessary third-party content to the game page:
` @import url('https://fonts.googleapis.com/css?family=Blinker&display=swap');`
All used content such as styles and fonts should be on the same host to avoid third-party tracking and increasing load times because of multiple hosts and connections.

samporapeli committed:
Change from remote font to local

samporapeli commented:
@ilarikuoppala third-party font service is no more used. Font is loaded from `web/assets/fonts`.

samporapeli committed:
Add support for custom fonts in canvas

Change Dark theme's font to Blinker

samporapeli committed:
Refactor changing themes logic

Code now uses object `themes` to store theme objects.
This commit removes messy if-statements from theme changing logic.

samporapeli committed:
Merge branch 'master' of github.com:ilarikuoppala/onlinesnake

samporapeli committed:
Move control buttons closer to game canvas

samporapeli commented:
@ilarikuoppala what do you think of this theming feature?
